src/frontend/src/handler/create_sql_function.rs

```diff
@@ -0,0 +1,180 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::anyhow;
+use arrow_schema::Fields;
+use itertools::Itertools;
+use pgwire::pg_response::StatementType;
+use risingwave_common::catalog::FunctionId;
+use risingwave_common::types::DataType;
+use risingwave_pb::catalog::function::{Kind, ScalarFunction, TableFunction};
+use risingwave_pb::catalog::Function;
+use risingwave_sqlparser::ast::{
+    CreateFunctionBody, FunctionDefinition, ObjectName, OperateFunctionArg,
+};
+use risingwave_udf::ArrowFlightUdfClient;
+
+use super::*;
+use crate::catalog::CatalogError;
+use crate::{bind_data_type, Binder};
+
+pub async fn handle_create_sql_function(
+    handler_args: HandlerArgs,
+    or_replace: bool,
+    temporary: bool,
+    name: ObjectName,
+    args: Option<Vec<OperateFunctionArg>>,
+    returns: Option<CreateFunctionReturns>,
+    params: CreateFunctionBody,
+) -> Result<RwPgResponse> {
+    if or_replace {
+        bail_not_implemented!("CREATE OR REPLACE FUNCTION");
+    }
+
+    if temporary {
+        bail_not_implemented!("CREATE TEMPORARY FUNCTION");
+    }
+
+    let language = "sql".to_string();
+    // Just a basic sanity check for language
+    if let Some(lang) = params.language {
+        let lang = lang.real_value().to_lowercase();
+        debug_assert!(lang == "sql", "`language` for sql udf function must be sql");
+    }
+
+    // SQL udf function supports both single quote (i.e., as 'select $1 + $2')
+    // and double dollar (i.e., as $$select $1 + $2$$) for as clause
+    let mut identifier = "".to_string();
+    if let Some(FunctionDefinition::SingleQuotedDef(id)) = params.as_.clone() {
+        identifier = id;
+    }
+    if let Some(FunctionDefinition::DoubleDollarDef(id)) = params.as_ {
+        identifier = id;
+    }
+    if identifier.is_empty() {
+        return Err(ErrorCode::InvalidParameterValue("AS must be specified".to_string()).into());
+    }
```

Comment:
rename `identifier` to function `body`?
```suggestion
    let body = match &params.as_ {
        Some(FunctionDefinition::SingleQuotedDef(s)) => s.clone(),
        Some(FunctionDefinition::DoubleDollarDef(s)) => s.clone(),
        None => return Err(ErrorCode::InvalidParameterValue("AS must be specified".to_string()).into()),
    };
```